sale_stock_picking_invoicing/models/stock_picking.py

```diff
@@ -0,0 +1,18 @@
+# Copyright (C) 2021-TODAY Akretion
+# @author Magno Costa <magno.costa@akretion.com.br>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _get_partner_to_invoice(self):
+        partner_id = super()._get_partner_to_invoice()
+        partner = self.env["res.partner"].browse(partner_id)
```

Comment:
you can avoid this browse here if you compare partner_id with self.sale_id.partner_invoice_id.id directly later (partner_invoice_id is required)